codalab/server/bundle_manager.py

```diff
@@ -347,29 +341,29 @@ def _schedule_run_bundles_on_workers(self, workers, staged_bundles_to_run):
                 # Get all the CodaLab's public workers
                 workers_list = workers.user_owned_workers(self._model.root_user_id)
 
-            workers_list = self._deduct_worker_resources(workers_list, uuid_to_running_bundles)
+            workers_list = self._deduct_worker_resources(workers_list, running_bundles_cache)
             workers_list = self._filter_and_sort_workers(workers_list, bundle, bundle_resources)
 
             # Try starting bundles on the workers that have enough computing resources
             for worker in workers_list:
-                if self._try_start_bundle(workers, worker, bundle):
+                if self._try_start_bundle(workers, worker, bundle, bundle_resources):
                     break
 
-    def _deduct_worker_resources(self, workers_list, uuid_to_running_bundles):
+    def _deduct_worker_resources(self, workers_list, running_bundles_cache):
         """
         From each worker, subtract resources used by running bundles. Modifies the list.
         """
 
         for worker in workers_list:
             for uuid in worker['run_uuids']:
                 # Verify if the current bundle exists in both the worker table and the bundle table
-                if uuid not in uuid_to_running_bundles:
+                if uuid not in running_bundles_cache:
                     logger.info(
                         'Bundle {} exists on worker {} but no longer found in the bundle table. '
                         'Skipping for resource deduction.'.format(uuid, worker['worker_id'])
                     )
                     continue
-                bundle_resources = self._compute_bundle_resources(uuid_to_running_bundles.get(uuid))
+                bundle_resources = running_bundles_cache.get(uuid).get("bundle_resource")
```

Comment:
should be `bundle_resources`, no?

Also, should just be `running_bundles_cache[uuid]` since the key is guaranteed to exist.

Reply:
Updated!